### PR TITLE
Fix for #813

### PR DESCRIPTION
**Pull request by pkukkon**
Changed URL name generation so that if the name looks like a file name, the unique counter is appended before the file extension rather than the end of the name.

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java [hunk 1]
```diff
@@ -887,6 +887,10 @@ public synchronized String generateUniqueUrlName(WorkspaceNode parent, String ti
 
   public synchronized String generateUniqueUrlName(WorkspaceNode parent, WorkspaceNode targetNode, String title) {
     String urlName = generateUrlName(title);
+    String fileName = StringUtils.substringBeforeLast(urlName, ".");
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java]: Wouldn't it be safer to use FilenameUtils.getBaseName?

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java [hunk 2]
```diff
@@ -887,6 +887,10 @@ public synchronized String generateUniqueUrlName(WorkspaceNode parent, String ti
...
+    String extension = StringUtils.substringAfterLast(urlName, ".");
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java]: Wouldn't it be safer to use FilenameUtils.getExtension?

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java [hunk 3]
```diff
@@ -901,7 +905,7 @@ public synchronized String generateUniqueUrlName(WorkspaceNode parent, Workspace
             break;
           }
           // uniqueName in use, try again with the next candidate (name, name-2, name-3, etc.)
-          uniqueName = urlName + "-" + ++i;
+          uniqueName = isFileName ? fileName + "-" + ++i + "." + extension : urlName + "-" + ++i;
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java]: Refactor to use String.format to make it more readable

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java (hunk 1 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java]: FilenameUtils.getBaseName is only suitable for dealing with file names. This code deals with titles (and occasional file names). In other words, if this method was given a string "1. luku", using FilenameUtils would turn it into name "1" and extension " luku", leading to unpredictable behaviour further down the line.

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java (hunk 2 above)
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceMaterialController.java]: See above.

**Pull request merged**